File: KnowledgeCenterClient/src/app/shared/models/BasePagination.ts
interface BasePagination {
  page: number;
  size: number;
}

export interface BasePaginationRequest<T> extends BasePagination {
  filters: T;
}

export interface BasePaginationResponse<T> extends BasePagination  {
  data: T;
  page: number;
  size: number;
  totalItems: number;
  totalPages: number;
}

File: KnowledgeCenterServer/KnowledgeCenter.Common/BasePagination.cs
namespace KnowledgeCenter.Common
{
    public class BasePagination
    {
        public int Page { get; set; } = 1;

        private int size = 100;
        public int Size { 
            get {
                return size;
            }
            set {
                if(value == 0) {
                    size = 10000;
                } else {
                    size = value;
                }
            }
        }
    }
}
